test/extensions/filters/http/ext_proc/mutation_utils_test.cc

```diff
@@ -115,7 +117,63 @@ TEST(MutationUtils, TestApplyMutations) {
       {"x-envoy-strange-thing", "No"},
   };
 
-  EXPECT_TRUE(TestUtility::headerMapEqualIgnoreOrder(headers, expected_headers));
+  EXPECT_THAT(&headers, HeaderMapEqualIgnoreOrder(&expected_headers));
+}
+
+// Ensure that we actually replace the body
+TEST(MutationUtils, TestBodyMutationReplace) {
+  Buffer::OwnedImpl buf;
+  TestUtility::feedBufferWithRandomCharacters(buf, 100);
+  BodyMutation mut;
+  mut.set_body("We have replaced the value!");
+  MutationUtils::applyBodyMutations(mut, buf);
+  EXPECT_EQ("We have replaced the value!", buf.toString());
+}
+
+// If an empty string is included in the "body" field, we should
+// replace the body with nothing
+TEST(MutationUtils, TestBodyMutationReplaceEmpty) {
+  Buffer::OwnedImpl buf;
+  TestUtility::feedBufferWithRandomCharacters(buf, 100);
+  BodyMutation mut;
+  mut.set_body("");
+  MutationUtils::applyBodyMutations(mut, buf);
+  EXPECT_EQ(0, buf.length());
+}
+
+// Clear the buffer if the "clear_buffer" flag is set
+TEST(MutationUtils, TestBodyMutationClearYes) {
+  Buffer::OwnedImpl buf;
+  TestUtility::feedBufferWithRandomCharacters(buf, 100);
+  BodyMutation mut;
+  mut.set_clear_body(true);
+  MutationUtils::applyBodyMutations(mut, buf);
+  EXPECT_EQ(0, buf.length());
+}
+
+// Don't clear the buffer if the "clear_buffer" flag is set to false,
+// which is weird, but possible
```

Comment:
Should it be possible? Or should we PGV validate that it's always true if set?

Reply:
Do you mean add validation to the protobuf so that a proto like that can't be created?  This may be paranoia on my part, but I assume that we can't guarantee that such validation will actually be run on the client so it's best to check.

Reply:
We run PGV validations against all protos on config ingestion, so if we put in a PGV annotation on a field, it will be validated before reaching your extension code.

Reply:
Yes, but these messages come from the ext_proc server, not from the config, so we have no control over who generated the protos.

Reply:
Got it. You could always call `Validate()` on the wire message when received, but you would need to handle errors in any case.